packages/gatekeeper/src/components/logout.tsx

```diff
@@ -4,22 +4,46 @@ import { connect } from 'react-redux';
 import { Redirect } from 'react-router-dom';
 import { LOGIN_URL } from '../helpers/constants';
 
-/** interface to describe props for Logout component */
+/** interface to describe props for Logout component
+ * @member {typeof logOutUser}logoutActionCreator action creator that logs out user.
+ * @member {string} redirectpath The URL we redirect to after loging out.
+ * @member {string} logoutURL optional parameter provided if/when
+ * you would like to logout of the authentication server.
+ */
 export interface LogoutProps {
   logoutActionCreator: typeof logOutUser;
   redirectPath: string;
+  logoutURL: string | null;
 }
 
 /** default props for Logout component */
 export const defaultLogoutProps: LogoutProps = {
   logoutActionCreator: logOutUser,
+  logoutURL: null,
   redirectPath: LOGIN_URL
 };
 
+/**
+ * Open another window and go to the logout URL.
+ * @param {string} logoutURL URL string representing the auth server logout URL endpoint.
+ */
+function logoutFromAuthServer(logoutURL: string) {
+  const logoutWindow: Window | null = window.open(logoutURL);
+  const timer: NodeJS.Timeout = setInterval(() => {
+    if (logoutWindow) {
+      logoutWindow.close();
+    }
+    clearInterval(timer);
+  }, 20);
+}
+
 /** Logout component */
 const Logout = (props: LogoutProps) => {
   const { logoutActionCreator, redirectPath } = props;
   logoutActionCreator();
+  if (props.logoutURL) {
+    logoutFromAuthServer(props.logoutURL);
```

Comment:
I think the usage here can be improved by making `logoutFromAuthServer` a prop that the LogOut component takes in.  The prop's signature is a function that takes in a string and outputs nothing (void).  Doing it this way has at least two advantages:
- it helps us keep this component in line with the [Open-Closed principle](https://stackify.com/solid-design-open-closed-principle/)
- it means that anyone who uses this component and wants to implement the logging out differently can do so easily

Reply:
That's something that should be done. since the component will now be open to extension, those who are able to use fetch(for example) could use fetch to logout.

Reply:
Cool.  Please do so.  I think you'll also need to move the function to a module in the helpers directory of this package.

Reply:
done!